src/script_interface/electrostatics/CoulombP3M.hpp

```diff
@@ -103,6 +114,33 @@ class CoulombP3M : public Actor<CoulombP3M<Architecture>, ::CoulombP3M> {
     m_tune = get_value<bool>(params, "tune");
     m_tune_timings = get_value<int>(params, "timings");
     m_tune_verbose = get_value<bool>(params, "verbose");
+    m_tune_limits = {std::nullopt, std::nullopt};
+    if (params.contains("tune_limits")) {
+      std::vector<Variant> range;
+      try {
+        auto const val = get_value<std::vector<int>>(params, "tune_limits");
+        assert(val.size() == 2u);
+        range.emplace_back(val[0u]);
+        range.emplace_back(val[1u]);
+      } catch (...) {
+        range = get_value<std::vector<Variant>>(params, "tune_limits");
+        assert(range.size() == 2u);
+      }
```

Comment:
A Python list containing only integers is passed to the script interface as a `std::vector<int>`, while a list containing at least one None is considered mixed type, and is passed as `std::vector<Variant>`. The `get_value()` function cannot convert a `std::vector<int>` object to a `std::vector<Variant>` object:

`RuntimeError: Provided argument of type 'std::vector<int>{.size=2}' for parameter 'tune_limits' is not convertible to 'std::vector<ScriptInterface::Variant>' because it contains a value that is not convertible to 'ScriptInterface::Variant'`

I've rewritten this code using a type check instead of a try/catch to make it more readable.